app/models/user.rb

```diff
@@ -10,8 +12,28 @@ class User < ApplicationRecord
   validates :password, presence: true, length: { minimum: 6 }
   has_secure_password
 
-  def User.digest(string)
-    cost = ActiveModel::SecurePassword.min_cost ? BCrypt::Engine::MIN_COST : BCrypt::Engine.cost
-    BCrypt::Password.create(string, cost: cost)
+  def remember
+    self.remember_token = User.new_token
+    update_attribute(:remember_digest, User.digest(remember_token))
```

Comment:
```suggestion
    update_columns(remember_digest: User.digest(remember_token))
```
Question: trong Rails có mấy loại method update, giống/khác nhau của mỗi loại?

Reply:
dạ có 6 loại update
- update: cập nhật thuộc tính và lưu vào db, trả về boolean (cần phải pass qua các validations ở model)
- update!: tương tự update nhưng nếu lưu không thành công thì sẽ bắn ra 1 ngoại lệ
-  update_attribute: cập nhật 1 thuộc tính và lưu trực tiếp vào db và bỏ qua các validations nhưng vẫn chạy callback
- update_attributes: tương tự như update và được dùng trước phiên bản Rails 6.x
- update_column: cập nhật 1 thuộc tính trực tiếp trong db mà không chạy validations, callback và không cập nhật lại timestamps
- update_columns: tương tự update_column nhưng nhiều cột thay thì 1

**so sánh**
- update, update! và update_attributes thì có chạy validations còn update_attribute, update_column, update_columns bỏ qua validations
- chỉ có update_column và update_columns bỏ qua callback và không cập nhật timestamps
- chỉ có mình update! là bắn ra ngoại lệ